### PR TITLE
GH-111435 sharing of bool type in sub interpreters

### PR DESCRIPTION
**Pull request by tonybaloney**
This PR adds sharing of bool type to the interpreters API

* Issue: gh-111435

### CODE REVIEW

**Comment by corona10** [comment]
```
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\cpython\cpython\Lib\test\test_interpreters.py", line 816, in test_not_shareable
    self.assertFalse(
AssertionError: True is not false
```

File: Python/pystate.c
Comment (Eclips4) [inline Python/pystate.c]: PEP 7

File: Python/pystate.c
Comment (ericsnowcurrently) [inline Python/pystate.c]: Ah, that looked like `011` and `111`, which was a little confusing.  Is the suffix necessary?

**Review by ericsnowcurrently — COMMENTED** [review]
Mostly LGTM

File: Python/pystate.c
Comment (ericsnowcurrently) [reply Python/pystate.c]: What's wrong specifically?

Comment (Eclips4) [reply Python/pystate.c]: Oh, I'm misread something. Sorry for the false report.

**Comment by ericsnowcurrently** [comment]
Sorry, I moved the registry code to Python/crossinterp.c.

**Comment by tonybaloney** [comment]
> Sorry, I moved the registry code to Python/crossinterp.c.

rebased

File: Python/pystate.c
Comment (tonybaloney) [reply Python/pystate.c]: MSVC would normally raise a warning about an int being cast to a `void*`. Maybe CPython has that warning disabled, I'll try it

Comment (tonybaloney) [reply Python/pystate.c]: llvm does as well

```default
Python/crossinterp.c:619:13: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
            (void *) (Py_IsTrue(obj) ? 1 : 0), NULL,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

Comment (tonybaloney) [reply Python/pystate.c]:
```cpp
            (void *) (Py_IsTrue(obj) ? (uintptr_t) 1 : (uintptr_t) 0), NULL,
```
would accomplish the same thing. I haven't seen many suffixes in the CPython source code so this is probably a better option

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

**Comment by ericsnowcurrently** [comment]
I'll merge this once you've had a chance to resolve the merge conflicts from the sharing-`float` PR.

**Comment by tonybaloney** [comment]
Rebased and conflicts resolved

**Comment by ericsnowcurrently** [comment]
Thanks, @tonybaloney!

**Pull request merged**
